src/main/java/com/devonfw/ide/sonarqube/common/impl/check/DevonUcImplSecurityConstraintCheck.java

```diff
@@ -0,0 +1,149 @@
+package com.devonfw.ide.sonarqube.common.impl.check;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.AnnotationTree;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.CompilationUnitTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.ModifierKeywordTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.TypeTree;
+
+import com.devonfw.ide.sonarqube.common.api.JavaType;
+
+/**
+ * {@link DevonArchitecture3rdPartyCheck} verifies that all Use-Case implementation methods are annotated with a
+ * security constraint from javax.annotation.security.
+ */
+@Rule(key = "E7", name = "Devon Uc Impl Security Constraint Check", //
+    priority = Priority.CRITICAL, tags = { "architecture-violation", "devonfw" })
+public class DevonUcImplSecurityConstraintCheck extends DevonArchitectureCheck {
```

Comment:
I agree that we should open a "category" for security.
* create a security sub-package
* start a new key Namespace - unfortunately we have already used `S` for scope. Should we change all Key-Prefixes to something more expressive? `SEC*` for security, `SCP*` for scope, `LYR*` for layer, `CMP*` for component, `3PY` for third-party... Otherwise we would need to use something strange for security like `Y`. After all it is just an ID...
* Issues triggered by security rules should become `Vulnerabilities` in SonarQube as done by other checks.
* Consider https://docs.sonarqube.org/latest/user-guide/security-rules/ and https://www.sonarqube.org/features/security/

Reply:
I will give it the key `Y1` for now. I agree that we could think about the names of keys being more expressive